nifi-docs/src/main/asciidoc/toolkit-guide.adoc

```diff
@@ -486,7 +486,22 @@ This protection scheme uses https://www.vaultproject.io/docs/secrets/kv/kv-v1[Ha
 
 ==== AZURE_KEYVAULT_KEY [[AZURE_KEYVAULT_KEY]]
-This protection scheme uses keys managed by https://docs.microsoft.com/en-us/azure/key-vault/keys/about-keys[Azure Key Vault Keys] for encryption and decryption. Azure Key Vault configuration properties can be stored in the `bootstrap-azure.conf` file, as referenced in the `bootstrap.conf` of NiFi or NiFi Registry. The provider will utilize the Azure default credentials provider chain as described in the https://docs.microsoft.com/en-us/java/api/overview/azure/security-keyvault-keys-readme?view=azure-java-stable[Azure Key Vault Key client library for Java] documentation. Therefore, when using the AZURE_KEYVAULT_KEY protection scheme, the `nifi(.registry)?.bootstrap.protection.azure.keyvault.conf` property in the `bootstrap.conf` specified using the `-b` flag must be available to the Encrypt Configuration Tool and must be configured as described in the <<administration-guide.adoc#_azure_key_vault_key_provider, Azure Key Vault Key provider>> section in the link:administration-guide.html[NiFi Administration Guide].
+This protection scheme uses keys managed by https://docs.microsoft.com/en-us/azure/key-vault/keys/about-keys[Azure Key Vault Keys] for encryption and decryption.
+Azure Key Vault configuration properties can be stored in the `bootstrap-azure.conf` file, as referenced in the `bootstrap.conf` of NiFi or NiFi Registry.
+The provider will utilize the Azure default credentials provider chain as described in the
+https://docs.microsoft.com/en-us/java/api/overview/azure/security-keyvault-keys-readme?view=azure-java-stable[Azure Key Vault Key client library for Java] documentation.
+Therefore, when using this protection scheme, the `nifi.bootstrap.protection.azure.keyvault.conf` property
+in the `bootstrap.conf` specified using the `-b` flag must be available to the Encrypt Configuration Tool
+and must be configured as described in the <<administration-guide.adoc#_azure_key_vault_key_provider, Azure Key Vault Key provider>> section in the link:administration-guide.html[NiFi Administration Guide].
+
+==== AZURE_KEYVAULT_SECRET [[AZURE_KEYVAULT_SECRET]]
+This protection scheme uses secrets managed by https://docs.microsoft.com/en-us/azure/key-vault/secrets/about-secrets[Azure Key Vault Secrets] for encryption and decryption.
+Azure Key Vault configuration properties can be stored in the `bootstrap-azure.conf` file, as referenced in the `bootstrap.conf` of NiFi or NiFi Registry.
+The provider will utilize the Azure default credentials provider chain as described in the
+https://docs.microsoft.com/en-us/java/api/overview/azure/security-keyvault-keys-readme?view=azure-java-stable[Azure Key Vault Key client library for Java] documentation.
```

Comment:
I think it could be useful to link out to https://docs.microsoft.com/en-us/java/api/overview/azure/identity-readme?view=azure-java-stable and/or https://docs.microsoft.com/en-us/java/api/com.azure.identity.defaultazurecredential?view=azure-java-stable

These include (IMO) a better explanation of how the chain works.

Also, I think it'd be useful to mention what the default is (e.g., no config it's going to us an managed identity).

Reply:
Thanks for the feedback @jfrazee! The first link for the Identity client seems to be the better one in terms of details. 

Without duplicating the description, do you think it is sufficient to describe the implementation in general by saying that it will attempt to reading from Azure-defined environment variables and system properties, with a fallback to managed identity credentials?

Reply:
@jfrazee I pushed an update that includes new references to the Azure Identity library and the Managed Identity authentication approach. I also streamlined the documentation to reduce duplication, hopefully the updates provide more straightforward details.